math/modulararithmetic/modularexponentiation.go

```diff
@@ -0,0 +1,42 @@
+package modulararithmetic
+
+import (
+	"errors"
+	"math"
+)
+
+// ErrorIntOverflow For asserting that the values do not overflow in Int64
+var ErrorIntOverflow = errors.New("Integer Overflow")
+
+// ModularExponentiation returns base^exponent % mod
+func ModularExponentiation(base, exponent, mod int64) int64 {
+	if mod == 1 {
+		return 0
+	}
+	_, err := Multiply64BitInt(mod-1, mod-1)
+
+	if err != nil {
+		panic(err)
```

Comment:
Code in this repo tends to avoid calling panic().

https://github.com/TheAlgorithms/Go/search?q=panic

Is that really the most elegant solution for production code?

Reply:
Are you talking about graceful error handling? for example:

```Go
func ModularExponentiation(base, exponent, mod int64) (int64, error) {
...
}

```